smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -98,14 +100,24 @@ contract MixinPurchase is
       emit RenewKeyPurchase(_recipient, newTimeStamp);
     }
 
+    
     (uint inMemoryKeyPrice, uint discount, uint tokens) = _purchasePriceFor(_recipient, _referrer, _data);
     if (discount > 0)
     {
-      unlockProtocol.recordConsumedDiscount(discount, tokens);
+      try unlockProtocol.recordConsumedDiscount(discount, tokens) 
+      {} 
```

Comment:
Hum? What's this? Some Solidity feature that I don't know about?

Reply:
Also, I think we can cleanup `recordConsumedDiscount` because it is not used... so let's simply remove it for now? (that'll save gas!)